src/contracts/Executor.sol

```diff
@@ -121,13 +130,22 @@ contract Executor is Context {
             if (isProposalExecuted(proposals[i].originDomainID, proposals[i].depositNonce)) {
                 continue;
             }
-            bytes32 stateRoot;
+            bytes32 expectedStateRoot;
             bytes32 storageRoot;
             address routerAddress = _originDomainIDToRouter[proposals[i].originDomainID];
 
-            IStateRootStorage stateRootStorage = IStateRootStorage(_securityModels[proposals[i].securityModel]);
-            stateRoot = stateRootStorage.getStateRoot(proposals[i].originDomainID, slot);
-            storageRoot = StorageProof.getStorageRoot(accountProof, routerAddress, stateRoot);
+            IStateRootStorage checkingStateRootStorage = IStateRootStorage(
+                _securityModels[proposals[i].securityModel][0]
+            );
+            expectedStateRoot = checkingStateRootStorage.getStateRoot(proposals[i].originDomainID, slot);
+
+            for (uint256 j = 1; j < _securityModels[proposals[i].securityModel].length; j++) {
+                IStateRootStorage stateRootStorage = IStateRootStorage(_securityModels[proposals[i].securityModel][j]);
+                bytes32 stateRoot = stateRootStorage.getStateRoot(proposals[i].originDomainID, slot);
+                if(expectedStateRoot != stateRoot) revert StateRootDoesNotMatch(stateRoot);
```

Comment:
Maybe it makes sense to add the verifier address or index to the error for debug purposes?